src/st-flash/flash_opts.c

```diff
@@ -233,22 +239,21 @@ int flash_get_opts(struct flash_opts* o, int ac, char** av) {
         av++;
     }
 
-    /* command and (optional) device name */
+    // command and (optional) device name
     while (ac >= 1) {
         if (strcmp(av[0], "erase") == 0) {
-            if (o->cmd != FLASH_CMD_NONE) { return(-1); }
-
+            if (FLASH_CMD_NONE != o->cmd) { return(-1); }
             o->cmd = FLASH_CMD_ERASE;
         } else if (strcmp(av[0], "read") == 0) {
-            if (o->cmd != FLASH_CMD_NONE) { return(-1); }
+            if (FLASH_CMD_NONE != o->cmd) { return(-1); }
 
             o->cmd = FLASH_CMD_READ;
         } else if (strcmp(av[0], "write") == 0) {
-            if (o->cmd != FLASH_CMD_NONE) { return(-1); }
+            if (FLASH_CMD_NONE != o->cmd) { return(-1); }
 
             o->cmd = FLASH_CMD_WRITE;
         } else if (strcmp(av[0], "reset") == 0) {
-            if (o->cmd != FLASH_CMD_NONE) { return(-1); }
+            if (FLASH_CMD_NONE != o->cmd) { return(-1); }
```

Comment:
Just to gain some knowledge out of this:
Does clang still have problems during compilation if you change the order back to the original state now?

Reply:
I reverted those too now, waiting for travis check

Reply:
The CI build state already confirms that there are no further issues. :tada: